corim/signedcorim.go

```diff
@@ -126,9 +129,45 @@ func (o *SignedCorim) FromCOSE(buf []byte) error {
 	return nil
 }
 
+// AddSigningCert adds a DER-encoded X.509 certificate to be included in the
+// protected header of the COSE Sign1 message as the leaf certificate in X5Chain.
+func (o *SignedCorim) AddSigningCert(der []byte) error {
+	if der == nil {
+		return errors.New("nil signing cert")
+	}
+
+	cert, err := x509.ParseCertificate(der)
+	if err != nil {
+		return fmt.Errorf("invalid signing certificate: %w", err)
+	}
+
+	o.signingCert = cert
+	return nil
+}
+
+// AddIntermediateCerts adds DER-encoded X.509 certificates to be included in the protected
+// header of the COSE Sign1 message as part of the X5Chain.
+// The certificates must be concatenated with no intermediate padding, as per X.509 convention.
```

Comment:
AFAIK it's a go crypto/x509 convention rather than a general convention around X.509 chain handling.

```suggestion
// The certificates must be concatenated with no intermediate padding.
```